src/Factory.php

```diff
@@ -59,6 +59,8 @@ class Factory
     /** @var callable[] */
     private array $afterPersist = [];
 
+    private bool $shouldPersist = true;
```

Comment:
I want to say no - but I can't remember why we couldn't use `$this->persist`. When I get some bandwidth this week, i'll dive back into this PR and see if we can't tidy things up..

Note to self:
run step debugger through https://github.com/zenstruck/foundry/blob/b66ed4ca2272491294928cb8ef6910c5c42c7954/src/Story.php#L175 to find out why `$this->persist` was a no-go

Reply:
don't bother with debugging or what, I think it will be okay to merge as-is. First, because as you said, it's a bug fix. And secondly, it kinda OK to not have a super perfect code in 1.x which in few months will be legacy

Reply:
I've tested your PR and used `Factory::$persist instead of the new property `Factory::$shouldPersist` and all tests are green :thinking: